frontend/routes/publishing/(_islands)/publishing.tsx

```diff
@@ -0,0 +1,189 @@
+// Copyright 2024 the JSR authors. All rights reserved. MIT license.
+import { computed, Signal, useSignalEffect } from "@preact/signals";
+import { useEffect } from "preact/hooks";
+import { api, path } from "../../../utils/api.ts";
+import {
+  PublishingTask,
+  PublishingTaskStatus,
+} from "../../../utils/api_types.ts";
+import { ErrorIcon } from "../../../components/icons/Error.tsx";
+import { Check } from "../../../components/icons/Check.tsx";
+import { Pending } from "../../../components/icons/Pending.tsx";
+
+export interface VersionPublishStatus {
+  loading: boolean;
+  task?: PublishingTask;
+}
+
+const statusVerb: Record<PublishingTaskStatus, string> = {
+  "pending": "is queued",
+  "success": "succeeded",
+  "failure": "failed",
+  "processed": "is processing",
+  "processing": "is processing",
+};
+
+export function PackagePublishStatus(props: {
+  name: string;
+  version: string;
+  date: string;
+  status: Signal<VersionPublishStatus>;
+}) {
+  useEffect(() => {
+    const [scope, name] = props.name.slice(1).split("/");
+    const version = props.version;
+    const date = new Date(props.date);
+    let cancel = false;
+
+    (async () => {
+      while (true) {
+        if (cancel) break;
+        const task = await poll(scope, name, version, date);
+        props.status.value = {
+          loading: false,
+          task,
+        };
+        if (task?.status === "success" || task?.status === "failure") break;
+        await new Promise((resolve) => setTimeout(resolve, 4000));
+      }
+    })().catch((e) => console.error("polling error", scope, name, version, e));
+
+    (() => cancel = true);
```

Comment:
Nit: Missing return statement. I think this should be:

```suggestion
    return () => (cancel = true);
```